documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -495,12 +492,14 @@ For instance, instead of a `KeyChar` event being raised every time a character i
 ```cs
 public readonly record struct KeyboardState
 (
-    string? Text,
+    InputReadOnlyList<char>? Text,
     KeyState Keys
 );
 ```
 
-`Text` contains the text typed on the keyboard since `IKeyboard.BeginInput`. This is cleared (set to `null`) when `IKeyboard.EndInput` is called, and will not be non-`null` again until another `IKeyboard.BeginInput` call.
+`Text` contains the characters typed on the keyboard since `IKeyboard.BeginInput`, and accounts for backspaces. This is cleared (set to `null`) when `IKeyboard.EndInput` is called, and will not be non-`null` again until another `IKeyboard.BeginInput` call. Given that `KeyChar` events are raised one character at a time, this property will update one character at a time to keep the state consistent with the event.
+
+**INFORMATIVE TEXT:** This is something we can optimize in `InputList` to not be allocatey, rest assured it is not acceptable to the Silk.NET team to allocate a new list for every character.
```

Comment:
Quick question, the idea here is to do something like allocating a single chunk of memory, so all `KeyboardState.Text` have the same "pointer" to it but see it with different length?

Reply:
Yes that is what we're alluding to in this informative text.